zerver/webhooks/slack/doc.md

```diff
@@ -1,34 +1,79 @@
-Get Zulip notifications from Slack for messages on your team's
-public channels!
+# Zulip Slack integration
 
-See also the [Slack-compatible webhook](/integrations/doc/slack_incoming).
+Get Zulip notifications for messages on your team's public Slack
+channels! You can choose to map each **Slack channel** either to a
+**Zulip topic** or to a **Zulip channel**.
 
-1. {!create-channel.md!}
+See also the [Zulip Slack incoming webhook integration][1].
+
+!!! warn ""
+
+    Using [Slack's legacy Outgoing Webhook service][5] is no longer
+    recommended. Follow these instructions to switch to the new
+    [Slack Events API][3].
+
+{start_tabs}
+
+1. To map each Slack channel to a Zulip topic, [create one channel][2]
+   you'd like to use for Slack notifications. Otherwise, for each public
+   Slack channel, [create a Zulip channel][2] with the same name.
 
 1. {!create-an-incoming-webhook.md!}
 
-1. {!generate-integration-url.md!}
+1. {!generate-webhook-url-basic.md!}
+
+    To map each Slack channel to a Zulip topic, make sure that the
+    **Send all notifications to a single topic** option is disabled
+    when generating the URL. Add `&channels_map_to_topics=1` to the
+    generated URL.
+
+    Otherwise, add `&channels_map_to_topics=0` to the generated URL.
+    Note that any Zulip channel you specified when generating the URL
+    will be ignored in this case.
```

Comment:
We can use the new `config_options` feature added in d1ff871523841eb15b1f29f67c16c8d66e4bb850 to greatly simplify this documentation, by having the widget generate the required field.

Reply:
Added this TODO to #30827 for tracking.